core/src/main/java/com/sequenceiq/cloudbreak/service/stack/RoleDependencyService.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import static java.util.stream.Collectors.toSet;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import com.google.api.client.util.Strings;
+
+/**
+ * Returns the list of Componenets on which the given component depends.
+ * roleDependencyMap is not an exhaustive list of dependencies.
+ * In any case where dependency is not defined, it will return  "UNDEFINED_DEPENDENCY"
+ * @param the component name for which we need to find the dependency.
+ * @return the list of components on which the input parameter depends.
+ */
+
+@Service
+public class RoleDependencyService {
+
+    private static List<String> noDependencies = List.of("NO_DEPENDENCIES");
+
+    private static List<String> undefinedDependency = List.of("UNDEFINED_DEPENDENCY");
+
+    private Map<String, List<String>> roleDependencyMap = Map.ofEntries(
+            Map.entry("NODEMANAGER", List.of("RESOURCEMANAGER")));
+
+    public List<String> getRoleDependency(String str) {
+        return roleDependencyMap.getOrDefault(str, undefinedDependency);
+    }
+
+    public String getConstantUndefinedDependency() {
```

Comment:
Separate public stataic final variables for UNDEFINED_DEPENDENCY, and NO_DEPENDENCIES ?